src/main.ts

```diff
@@ -68,13 +66,12 @@ export default class CopilotPlugin extends Plugin {
     // Always have one instance of sharedState and chainManager in the plugin
     this.sharedState = new SharedState();
 
-    this.vectorStoreManager = new VectorStoreManager(this.app);
-    await this.vectorStoreManager.waitForInitialization();
```

Comment:
no need to explicitly block onload. The initialization will happen in the background and all public methods will wait for initialization.